src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -0,0 +1,271 @@
+                    <div style={{display:"inline-block", width:300, marginLeft:4, marginRight:4}}>
+                        <Select
+                            name="data-set-select"
+                            value={this.selectedMolecularProfile.molecularProfileId}
+                            onChange={this.onSelectDataSet}
+                            options={options}
+                            searchable={false}
+                            clearable={false}
+                            className="coexpression-select-profile"
+                        />
+                    </div>
+                    <InfoIcon style={{marginLeft:3}} tooltip={<span>{this.selectedMolecularProfile.description}</span>}/>
+                </div>
+            );
+        } else {
+            return <LoadingIndicator isLoading={true}/>;
+        }
+    }
+
+    private header() {
+        return (
+            <div>
+                {this.dataSetSelector}
+            </div>
+        );
+    }
+
+    private geneTabs() {
+        if (this.selectedMolecularProfile && this.selectedEntrezGeneId !== undefined) {
+            return (
+                <div>
+                    <MSKTabs
+                        id="coexpressionTabGeneTabs"
+                        activeTabId={this.selectedEntrezGeneId + ""}
+                        onTabClick={this.onSelectGene}
+                        className="coexpressionTabGeneTabs"
+                        unmountOnHide={true}
+                    >
+                        {this.props.genes.map((gene:Gene, i:number)=>{
+                            return (
+                                <MSKTab
+                                    className={`coExpressionTab${i}`}
+                                    key={i}
+                                    id={gene.entrezGeneId+""}
+                                    linkText={gene.hugoGeneSymbol}
+                                >
+                                </MSKTab>
+                            );
+                        })}
+                    </MSKTabs>
+                    <CoExpressionViz
+                        coExpressionCache={this.coExpressionCache}
+                        allDataRequested={this.isAllDataRequestedForCurrentTab()}
+                        setHighlightedCoExpression={this.setHighlightedCoExpressionForCurrentTab}
+                        highlightedCoExpression={this.getHighlightedCoExpressionForCurrentTab()}
+                        gene={this.props.genes.find(g=>(g.entrezGeneId === this.selectedEntrezGeneId))!}
+                        molecularProfile={this.selectedMolecularProfile!}
+                        numericGeneMolecularDataCache={this.props.numericGeneMolecularDataCache}
+                        mutationCache={this.hasMutationData ? this.props.mutationCache : undefined}
+                        onClickRequestAllData={this.requestAllData}
+                    />
+                </div>
+            );
+        } else {
```

Comment:
we can assume loading if !this.selectedMolecularProfile  ?

Reply:
yes, because that cant be undefined unless the available molecular profiles are not yet downloaded